tough/src/editor.rs

```diff
@@ -0,0 +1,660 @@
+    /// are used, the appropriate files are prefixed with their `sha256`.
+    pub fn write<P>(&self, outdir: P) -> Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        let consistent_snapshot = self.root.signed.signed.consistent_snapshot;
+        self.root.write(&outdir, consistent_snapshot)?;
+        self.targets.write(&outdir, consistent_snapshot)?;
+        self.snapshot.write(&outdir, consistent_snapshot)?;
+        self.timestamp.write(&outdir, consistent_snapshot)
+    }
+
+    /// Crawls a given directory and symlinks any targets found to the given
+    /// "out" directory.
+    ///
+    /// For each file found in the `indir`, the method gets the filename and
+    /// if the filename exists in `Targets`, the file's sha256 is compared
+    /// against the data in `Targets`. If this data does not match, the
+    /// method will fail. If all is well, the taret is symlinked.
+    pub fn link_targets<P1, P2>(&self, indir: P1, outdir: P2) -> Result<()>
+    where
+        P1: AsRef<Path>,
+        P2: AsRef<Path>,
+    {
+        std::fs::create_dir_all(&outdir).context(error::DirCreate {
+            path: &outdir.as_ref(),
+        })?;
+        // Get the absolute path of the indir and outdir
+        let abs_indir = std::fs::canonicalize(&indir).context(error::AbsolutePath {
+            path: indir.as_ref().to_owned(),
+        })?;
+        let abs_outdir = std::fs::canonicalize(&outdir).context(error::AbsolutePath {
+            path: outdir.as_ref().to_owned(),
+        })?;
+        let repo_targets = &self.targets.signed.signed.targets;
+
+        // Walk the absolute path of the indir. Using the absolute path here
+        // means that `entry.path()` call will return its absolute path.
+        let walker = WalkDir::new(abs_indir).follow_links(false);
+        for entry in walker {
+            let entry = entry.context(error::WalkDir)?;
+
+            // If the entry is a file...
+            if entry.file_type().is_file() {
+                // get the filename...
+                let file_name = entry
+                    .file_name()
+                    .to_str()
+                    .context(error::PathUtf8 { path: entry.path() })?;
+
+                // use the file name to see if a target exists in the repo
+                // with that name. If so...
+                if let Some(repo_target) = repo_targets.get(file_name) {
+                    // create a Target object using the entry's path, and then...
+                    let target_from_path = Target::from_path(entry.path())
+                        .context(error::TargetFromPath { path: entry.path() })?;
+
+                    // compare the hashes of the target from the repo and the
+                    // target we just created. If they are the same, this must
+                    // be the same file, symlink it.
+                    if target_from_path.hashes.sha256 == repo_target.hashes.sha256 {
```

Comment:
Similar to above suggestions, remove some nesting:
```suggestion
                    ensure!(target_from_path.hashes.sha256 == repo_target.hashes.sha256,
                        error::HashMismatch { bla });
```